### PR TITLE
[update] debugging

### PR DESCRIPTION
**Pull request by isdanni**
# file changed

- `debugging.rst`

# Related issue

#100

### CODE REVIEW

**Comment by isdanni** [comment]
@archibate @ArkhamWJZ @rexwangcc 

If anyone is available, feel free to help review!
(cannot request reviewer lol

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 1]
```diff
@@ -3,37 +3,40 @@
...
 msgid "Debugging"
-msgstr ""
+msgstr "调试"
 
 #: ../../debugging.rst:4
 msgid ""
 "Debugging a parallel program is not easy, so Taichi provides builtin "
 "utilities that could hopefully help you debug your Taichi program."
 msgstr ""
+"调试并行程序并不容易，因此Taichi提供了内置程序，希望能帮助你调试Taichi程"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"调试并行程序并不容易，因此Taichi提供了内置的支持，希望能帮助你调试Taichi程"
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 2]
```diff
@@ -3,37 +3,40 @@
...
+"序。"
 
 #: ../../debugging.rst:8
 msgid "Run-time ``print`` in kernels"
-msgstr ""
+msgstr "内核中的运行时\"打印“"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
msgstr "内核中的运行时 ``print``"
```
A word embraced by \` could means it's a Python keyword :)

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 3]
```diff
@@ -3,37 +3,40 @@
...
 
 #: ../../debugging.rst:12
 msgid "Debug your program with ``print()`` in Taichi-scope. For example:"
-msgstr ""
+msgstr "在Taichi范围内用 ``print()`` 调试程序。例如："
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
msgstr "在 Taichi 作用域内用 ``print()`` 调试程序。例如："
```
Taichi-scope, please see wiki for trans-table.

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 4]
```diff
@@ -69,6 +95,9 @@ msgid ""
 "little different from ``print`` in Python-scope. Please see details "
 "below."
 msgstr ""
+"目前，Taichi范围的 ``print`` 支持字符串、标量、矢量和矩阵表达式作为参数。"
+"Taichi范围中的 ``print`` 可能与 Python 范围中的 ``print`` 略有不同。请参"
+"阅下面的详细信息。"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"目前，Taichi 作用域的 ``print`` 支持字符串、标量、矢量和矩阵表达式作为参数。"
"Taichi 作用域中的 ``print`` 可能与 Python 作用域中的 ``print`` 略有不同。请参"
"阅下面的详细信息。"
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 5]
```diff
@@ -212,29 +301,44 @@ msgid ""
...
 "When you are done with debugging, simply set ``debug=False``. Now "
 "``assert`` will be ignored and there will be no runtime overhead."
 msgstr ""
+"完成调试后，只需设置 ``debug=False`` 。现在， ``assert`` 将被忽略，并且没"
+"有运行时开销。"
 
 #: ../../debugging.rst:154
 msgid "Compile-time ``ti.static_assert``"
-msgstr ""
+msgstr "编译时间 ``ti.static_assert``"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
msgstr "编译时 ``ti.static_assert``"
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 6]
```diff
@@ -144,10 +196,23 @@ msgid ""
...
+"    print('a[0] = ', a[0]) # 正确\n"
+"    print(f'a[0] = {a[0]}') # 错误，f-string 不被支持\n"
+"    print(\"a[0] = %f\" % a[0]) # 错误， 格式化字符串不被支持\n"
+"\n"
+"foo()"
 
 #: ../../debugging.rst:99
 msgid "Compile-time ``ti.static_print``"
-msgstr ""
+msgstr "编译时间 ``ti.static_print``"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
msgstr "编译时 ``ti.static_print``"
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 7]
```diff
@@ -156,6 +221,9 @@ msgid ""
 "provide ``ti.static_print`` to print compile-time constants. It is "
 "similar to Python-scope ``print``."
 msgstr ""
+"有时，在Taichi范围中打印 Python 范围的对象和常量（如数据类型或 SNodes）非"
+"常有用。因此，类似于 ``ti.static`` ，我们提供 ``ti.static_print`` 来打印"
+"编译时常数。它类似于 Python 范围的 ``print`` 。"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"有时，在Taichi 作用域中打印 Python 作用域的对象和常量（如数据类型或 SNodes）非"
"常有用。因此，类似于 ``ti.static`` ，我们提供 ``ti.static_print`` 来打印"
"编译时常数。它类似于 Python 作用域的 ``print`` 。"
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 8]
```diff
@@ -246,10 +350,17 @@ msgid ""
...
+"    ti.static_assert(dst.shape == src.shape, \"copy() needs src and dst "
+"tensors to be same shape\")\n"
+"    for I in ti.grouped(src):\n"
+"        dst[I] = src[I]\n"
+"    return x % 2 == 1"
 
 #: ../../debugging.rst:176
 msgid "Pretty Taichi-scope traceback"
-msgstr ""
+msgstr "漂亮的Taichi范围回溯"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
msgstr "美化 Taichi 作用域的栈回溯"
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 9]
```diff
@@ -258,6 +369,8 @@ msgid ""
 "from **Taichi-scope** could be extremely complicated and hard to read. "
 "For example:"
 msgstr ""
+"我们都知道，Python 提供了一个有用的堆栈回溯系统，它可以帮你轻松找到问题。"
+"但有时 **Taichi范围** 的堆栈回溯可能极其复杂且难以阅读。例如："
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"但有时 **Taichi  作用域** 的堆栈回溯可能极其复杂且难以阅读。例如："
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 10]
```diff
@@ -330,27 +463,80 @@ msgid ""
...
+"AssertionError"
 
 #: ../../debugging.rst:246
 msgid ""
 "You may already feel brain fried by the annoying ``decorated``'s and "
 "``__call__``'s. These are the Taichi internal stack frames. They have "
 "almost no benefit for end-users but make the traceback hard to read."
 msgstr ""
+"你可能已经觉得看 ``decorated`` 和 ``__call__`` 已经快大脑炸了 。这些是"
+"Taichi内部堆叠框架。它们对终端用户几乎没有好处，而且使回溯很难阅读。"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"你可能已经觉得看 ``decorated`` 和 ``__call__`` 大脑快炸了 。这些是"
"Taichi内部堆栈帧。它们对普通用户几乎没有好处，而且使回溯很难阅读。"
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 11]
```diff
@@ -330,27 +463,80 @@ msgid ""
...
 
 #: ../../debugging.rst:250
 msgid ""
 "For this purpose, we may want to use ``ti.init(excepthook=True)``, which "
 "*hooks* on the exception handler, and make the stack traceback from "
 "Taichi-scope easier to read and intuitive. e.g.:"
 msgstr ""
+"为此，我们可能希望使用 ``ti.init(excepthook=True)`` ，这在异常处理程序上 "
+"*挂钩* ，使Taichi范围中的堆栈回溯更易于阅读和直观。例如："
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"*挂钩* ，使 Taichi 作用域中的堆栈回溯更易于阅读和直观。例如："
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 12]
```diff
@@ -417,54 +607,115 @@ msgid ""
...
+"--------------------------------------------\n"
+"AssertionError"
 
 #: ../../debugging.rst:318
 msgid ""
 "See? Our exception hook has removed some useless Taichi internal frames "
 "from traceback. What's more, although not visible in the doc, the output "
 "is **colorful**!"
 msgstr ""
+"看到？我们的异常挂钩已经从回溯中删除了一些无用的Taichi内部框架。更重要的"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"看到了吧？我们的异常挂钩已经从回溯中删除了一些无用的 Taichi 内部框架。更重要的"
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 13]
```diff
@@ -417,54 +607,115 @@ msgid ""
...
+"是，虽然在文档中不可见，但输出是 **彩色** 的！"
 
 #: ../../debugging.rst:325
 msgid ""
 "For IPython / Jupyter notebook users, the IPython stack traceback hook "
 "will be overriden by the Taichi one when ``ti.enable_excepthook()``."
 msgstr ""
+"对于 IPython / Jupyter notebook 的用户，当 ``ti.enable_excepthook()`` 触"
+"发时，IPython 堆栈回溯挂钩将被Taichi的重挂。"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"对于 IPython / Jupyter notebook 的用户，当 ``ti.enable_excepthook()`` 触"
"发时，IPython 原有的堆栈回溯挂钩将被 Taichi 取代。"
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 14]
```diff
@@ -417,54 +607,115 @@ msgid ""
...
 msgid ""
-"Debugging a Taichi program can be hard even with the builtin tools above."
-" Here we showcase some common bugs that one may encounter in a Taichi "
-"program."
+"Debugging a Taichi program can be hard even with the builtin tools "
+"above. Here we showcase some common bugs that one may encounter in a "
+"Taichi program."
 msgstr ""
+"即使有上面的内置工具，调试太极程序也可能会很难。在这里，我们展示了一些"
+"Taichi程序中可能会遇到的常见错误。"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"即使有上面的内置工具，调试 Taichi 程序也可能会很难。在这里，我们展示了一些"
" Taichi 程序中可能会遇到的常见错误。"
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 15]
```diff
@@ -417,54 +607,115 @@ msgid ""
...
 msgid ""
 "Python code in Taichi-scope is translated into a statically typed "
-"language for high performance. This means code in Taichi-scope can have a"
-" different behavior compared with that in Python-scope, especially when "
+"language for high performance. This means code in Taichi-scope can have "
+"a different behavior compared with that in Python-scope, especially when "
 "it comes to types."
 msgstr ""
+"Taichi范围中的 Python 代码被翻译成静态类型语言以实现高性能。这意味着"
+"Taichi范围中的代码与 Python 范围中的代码可以有不同的行为，尤其是在类型方"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"Taichi 作用域中的代码与 Python 作用域中的代码可以有不同的行为，尤其是在类型方"
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 16]
```diff
@@ -521,18 +794,25 @@ msgid ""
...
 msgstr ""
+"```RuntimeError: [verify.cpp:basic_verify@40] stmt 8 cannot have operand "
+"7.```"
 
 #: ../../debugging.rst:388
 msgid ""
 "You may use ``ti.init(advance_optimization=False)`` to turn off advanced "
 "optimization and see if the issue still exists:"
 msgstr ""
+"你可以使用 ``ti.init(advance_optimization=False)`` 关闭高级优化，并查看问"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"你可以使用 ``ti.init(advanced_optimization=False)`` 关闭高级优化，并查看问"
```
We found an typo in en doc...

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 17]
```diff
@@ -542,12 +822,18 @@ msgid ""
 "\n"
 "..."
 msgstr ""
+"import taichi as ti\n"
+"\n"
+"ti.init(advance_optimization=False)\n"
```
Comment (archibate) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"ti.init(advanced_optimization=False)\n"
```

**Review by archibate — COMMENTED** [review]
LGTMig! I left a few nits.

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 18]
```diff
@@ -417,54 +607,115 @@ msgid ""
...
+"Taichi范围中的 Python 代码被翻译成静态类型语言以实现高性能。这意味着"
+"Taichi 作用域中的代码与 Python 作用域中的代码可以有不同的行为，尤其是在类型方"
+"面。"
 
 #: ../../debugging.rst:340
 msgid ""
 "The type of a variable is simply **determined at its initialization and "
 "never changes later**."
-msgstr ""
+msgstr "变量的类型只是 **在初始化时确定，以后永远不会更改** 。"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
msgstr "变量的类型只 **在初始化时确定，并且之后不会做更改** 。"
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 19]
```diff
@@ -417,54 +607,115 @@ msgid ""
...
 #: ../../debugging.rst:338
 msgid ""
 "Python code in Taichi-scope is translated into a statically typed "
-"language for high performance. This means code in Taichi-scope can have a"
-" different behavior compared with that in Python-scope, especially when "
+"language for high performance. This means code in Taichi-scope can have "
+"a different behavior compared with that in Python-scope, especially when "
 "it comes to types."
 msgstr ""
+"Taichi范围中的 Python 代码被翻译成静态类型语言以实现高性能。这意味着"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"Taichi 作用域中的 Python 代码被翻译成静态类型语言以实现高性能。这意味着"
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 20]
```diff
@@ -112,19 +158,25 @@ msgid ""
...
+"sync()``."
+msgstr "请注意，主机访问或程序终止也将隐式触发 ``ti.sync()``.。"
 
 #: ../../debugging.rst:79
 msgid ""
 "Note that ``print`` in Taichi-scope can only receive **comma-separated "
 "parameter**. Neither f-string nor formatted string should be used. For "
 "example:"
 msgstr ""
+"请注意，Taichi范围中的 ``print`` 只能接收 **逗号分隔参数**。不应使用 f 字"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"请注意，Taichi 作用域中的 ``print`` 只能接收 **逗号分隔参数**。不应使用 f 字"
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 21]
```diff
@@ -144,10 +196,23 @@ msgid ""
...
+"ti.init(arch=ti.cpu)\n"
+"a = ti.var(ti.f32, 4)\n"
+"\n"
+"\n"
+"@ti.kernel\n"
+"def foo():\n"
+"    a[0] = 1.0\n"
+"    print('a[0] = ', a[0]) # 正确\n"
+"    print(f'a[0] = {a[0]}') # 错误，f-string 不被支持\n"
+"    print(\"a[0] = %f\" % a[0]) # 错误， 格式化字符串不被支持\n"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"    print(\"a[0] = %f\" % a[0]) # 错误， 不支持格式化字符串\n"
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 22]
```diff
@@ -144,10 +196,23 @@ msgid ""
...
+"import taichi as ti\n"
+"ti.init(arch=ti.cpu)\n"
+"a = ti.var(ti.f32, 4)\n"
+"\n"
+"\n"
+"@ti.kernel\n"
+"def foo():\n"
+"    a[0] = 1.0\n"
+"    print('a[0] = ', a[0]) # 正确\n"
+"    print(f'a[0] = {a[0]}') # 错误，f-string 不被支持\n"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"    print(f'a[0] = {a[0]}') # 错误，不支持 f-string \n"
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 23]
```diff
@@ -175,30 +243,51 @@ msgid ""
...
 
 #: ../../debugging.rst:130
 msgid ""
 "Programmers may use ``assert`` statements in Taichi-scope. When the "
-"assertion condition failed, a ``RuntimeError`` will be raised to indicate"
-" the error."
+"assertion condition failed, a ``RuntimeError`` will be raised to "
+"indicate the error."
 msgstr ""
+"程序员可以在Taichi范围内使用 ``assert`` 语句。当断言条件失败时， "
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"程序员可以在 Taichi 作用域内使用 ``assert`` 语句。当断言条件失败时， "
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 24]
```diff
@@ -175,30 +243,51 @@ msgid ""
...
+"``RuntimeError`` 将被触发以指示错误。"
 
 #: ../../debugging.rst:133
 msgid ""
 "To make ``assert`` work, first make sure you are using the **CPU "
 "backend**. For performance reason, ``assert`` only works when ``debug`` "
 "mode is on, For example:"
 msgstr ""
+"若要使 ``assert`` 正常工作，首先请确保使用 **CPU 后端** 。出于性能原因， "
+"``assert`` 仅在 ``debug`` 模式处于打开状态时才有效，例如："
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"若要使 ``assert`` 正常工作，首先请确保使用 **CPU 后端** 运行程序。其次出于性能方面的考量， "
"``assert`` 仅在 ``debug`` 模式开启时有效，例如："
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 25]
```diff
@@ -212,29 +301,44 @@ msgid ""
...
+"        assert x[i] >= 0\n"
+"        x[i] = ti.sqrt(x)"
 
 #: ../../debugging.rst:149
 msgid ""
 "When you are done with debugging, simply set ``debug=False``. Now "
 "``assert`` will be ignored and there will be no runtime overhead."
 msgstr ""
+"完成调试后，只需设置 ``debug=False`` 。现在， ``assert`` 将被忽略，并且没"
+"有运行时开销。"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"完成调试后，只需设置 ``debug=False`` 。此时， ``assert`` 将被忽略，并且不会产生"
"运行时开销。"
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 26]
```diff
@@ -212,29 +301,44 @@ msgid ""
...
+"Like ``ti.static_print``, we also provide a static version of "
+"``assert``: ``ti.static_assert``. It can be useful to make assertions on "
+"data types, dimensionality, and shapes. It works whether ``debug=True`` "
+"is specified or not. When an assertion fails, it will raise an "
+"``AssertionError``, just like a Python-scope ``assert``."
 msgstr ""
+"与 ``ti.static_print`` 一样，我们还提供了 ``assert`` 的静态版本： ``ti."
+"static_assert`` 。对数据类型、维度和形状进行断言可能很有用。无论是否指定 "
+"``debug=True`` ，它都有效。当断言失败时，它将引发一个 "
+"``AssertionError`` ，就像Python范围的 ``assert`` 一样。"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"``AssertionError`` ，就像 Python 作用域中的 ``assert`` 一样。"
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 27]
```diff
@@ -246,10 +350,17 @@ msgid ""
...
+"    ti.static_assert(dst.shape == src.shape, \"copy() needs src and dst "
+"tensors to be same shape\")\n"
+"    for I in ti.grouped(src):\n"
+"        dst[I] = src[I]\n"
+"    return x % 2 == 1"
 
 #: ../../debugging.rst:176
 msgid "Pretty Taichi-scope traceback"
-msgstr ""
+msgstr "美化 Taichi 作用域的栈回溯"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
msgstr "优雅的 Taichi 作用域的栈回溯"
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 28]
```diff
@@ -258,6 +369,8 @@ msgid ""
 "from **Taichi-scope** could be extremely complicated and hard to read. "
 "For example:"
 msgstr ""
+"我们都知道，Python 提供了一个有用的堆栈回溯系统，它可以帮你轻松找到问题。"
+"但有时 **Taichi  作用域** 的堆栈回溯可能极其复杂且难以阅读。例如："
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"但有时 **Taichi 作用域** 的堆栈回溯(stack traceback)日志可能极其复杂且难以阅读。例如："
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 29]
```diff
@@ -330,27 +463,80 @@ msgid ""
...
+"AssertionError"
 
 #: ../../debugging.rst:246
 msgid ""
 "You may already feel brain fried by the annoying ``decorated``'s and "
 "``__call__``'s. These are the Taichi internal stack frames. They have "
 "almost no benefit for end-users but make the traceback hard to read."
 msgstr ""
+"你可能已经觉得看 ``decorated`` 和 ``__call__`` 大脑快炸了 。这些是"
+"Taichi内部堆栈帧。它们对普通用户几乎没有好处，而且使回溯很难阅读。"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"分析诸如 ``decorated`` 和 ``__call__`` 之类晦涩的信息也许能让你的大脑过载着火。其实这些是"
"Taichi 内部堆栈帧。直接暴露它们对普通用户几乎没有好处，并且会使回溯日志很难阅读。"
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 30]
```diff
@@ -330,27 +463,80 @@ msgid ""
...
 
 #: ../../debugging.rst:250
 msgid ""
 "For this purpose, we may want to use ``ti.init(excepthook=True)``, which "
 "*hooks* on the exception handler, and make the stack traceback from "
 "Taichi-scope easier to read and intuitive. e.g.:"
 msgstr ""
+"为此，我们可能希望使用 ``ti.init(excepthook=True)`` ，这在异常处理程序上 "
+"*挂钩* ，使 Taichi 作用域中的堆栈回溯更易于阅读和直观。例如："
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"为此，我们可能希望使用 ``ti.init(excepthook=True)`` ，这会与异常处理程序 "
"*挂钩(hook)* ，从而使 Taichi 作用域中的堆栈回溯日志更直观且易于阅读。例如："
```

File: locales/zh_CN/LC_MESSAGES/debugging.po [hunk 31]
```diff
@@ -417,54 +607,115 @@ msgid ""
...
+"AssertionError"
 
 #: ../../debugging.rst:318
 msgid ""
 "See? Our exception hook has removed some useless Taichi internal frames "
 "from traceback. What's more, although not visible in the doc, the output "
 "is **colorful**!"
 msgstr ""
+"看到了吧？我们的异常挂钩已经从回溯中删除了一些无用的 Taichi 内部框架。更重要的"
+"是，虽然在文档中不可见，但输出是 **彩色** 的！"
```
Comment (ArkhamWJZ) [inline locales/zh_CN/LC_MESSAGES/debugging.po]:
```suggestion
"看到了吧？我们的异常挂钩(exception hook)已经从回溯中删除了一些无用的 Taichi 内部堆栈帧。更重要的"
"是，虽然在文档中不可见，但这些输出都是 **彩色** 的！"
```

**Review by ArkhamWJZ — COMMENTED** [review]
A few nits, please feel free to update 🙆‍♂️

**Review by ArkhamWJZ — APPROVED** [review]

**Review by archibate — APPROVED** [review]
All LGTM!

**Pull request merged**